funnel/cli/periodic.py

```diff
@@ -29,6 +30,39 @@ class DataSource:
     datecolumn: Any
 
 
+def matomo_data():
+    tz = pytz.timezone('Asia/Kolkata')
+    now = utcnow().astimezone(tz)
+    yesterday = (now - relativedelta(days=1)).strftime('%Y-%m-%d')
+    referrers = f'https://matomo.hasgeek.tech?module=API&method=Referrers.getWebsites&idSite=1&period=day&date={yesterday}&filter_limit=-1&format=json&token_auth={app.config["MATOMO_TOKEN"]}'
+    socials = f'https://matomo.hasgeek.tech?module=API&method=Referrers.getSocials&idSite=1&period=day&date={yesterday}&filter_limit=-1&format=json&token_auth={app.config["MATOMO_TOKEN"]}'
+    pages = f'https://matomo.hasgeek.tech?module=API&method=Actions.getPageTitles&idSite=1&period=day&date={yesterday}&filter_limit=-1&format=json&token_auth={app.config["MATOMO_TOKEN"]}'
+
+    referrers = requests.get(referrers, timeout=5).json()
+    socials = requests.get(socials, timeout=5).json()
+    pages = requests.get(pages, timeout=5).json()
+
+    referrers_list = [(k['label'], k['nb_visits']) for k in referrers]
+    socials_list = [(k['label'], k['nb_visits']) for k in socials]
+    pages_list = [(k['label'], k['nb_visits']) for k in pages]
+
+    message = ""
+
+    message += "\n*Top Pages:*\n"
+    for index, i in enumerate(itertools.islice(pages_list, 7), 1):
```

Comment:
Why is `itertools.islice` needed here? It does the same thing as `pages_list[:7]`